rust-runtime/aws-smithy-http-server/src/routing/into_make_lambda_service.rs

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use http::uri;
+use lambda_http::{Request, RequestExt};
+use std::{
+    fmt::Debug,
+    task::{Context, Poll},
+};
+use tower::Service;
+
+/// A [`MakeService`] that produces AWS Lambda compliant services.
+///
+/// [`MakeService`]: tower::make::MakeService
+#[derive(Debug, Clone)]
+pub struct IntoMakeLambdaService<S> {
+    service: S,
+}
+
+impl<S> IntoMakeLambdaService<S> {
+    pub fn new(service: S) -> Self {
+        Self { service }
+    }
+}
+
+impl<S> Service<Request> for IntoMakeLambdaService<S>
+where
+    S: Service<Request>,
+{
+    type Error = S::Error;
+    type Response = S::Response;
+    type Future = S::Future;
```

Comment:
seems like this is where we might want to do the body conversion?

Reply:
Changes were made in order to avoid doing a body conversion from a `lambda_http::Body` to a `hyper::Body` it's unfortunate that we would have to do a body conversion here anyway for `ByteStream`?

Reply:
Indeed, it's unfortunate. There's an issue to make `ByteStream` work with all body types: https://github.com/awslabs/smithy-rs/issues/1125 but we haven't been able to complete it.